src/NuGet.Core/NuGet.Common/Telemetry/TelemetryEvent.cs

```diff
@@ -8,36 +8,36 @@ namespace NuGet.Common
     /// <summary> Represents a NuGet telemetry event data to pass to telemetry service. </summary>
     public class TelemetryEvent
     {
-        private IDictionary<string, object> _properties;
-        private IDictionary<string, object> _piiProperties;
+        private IDictionary<string, object?> _properties;
+        private IDictionary<string, object?> _piiProperties;
 
         /// <summary> Creates a new instance of <see cref="TelemetryEvent"/>. </summary>
         /// <param name="eventName"> Event name. </param>
-        public TelemetryEvent(string eventName) :
-            this(eventName, new Dictionary<string, object>())
+        public TelemetryEvent(string? eventName) :
```

Comment:
you mentioned some places create `null` events. Can that be changed so we can have a more realistic annotation this type?